sections/featured-blog.liquid

```diff
@@ -1,15 +1,124 @@
+{%-comment-%}
+From Vivian: I added lines 21, 22, 24, 25, 33, 34, 118, and 120 through 206, and I changed lines 
+32 and 120 (which I initially added via copy-paste) to be limited by my `postLimit` variable instead of by `section.settings.post_limit`.
+
+My thinking was that I had to account for the following possibilities: 
+1. There would be no pinned post. In this scenario, I skip the pinned post rendering section.
+2. There would be a pinned post. In this scenario, I begin with the pinned post rendering section and then follow by rendering 2 non-pinned posts.
+3. There would be more than one post tagged "pinned." In this scenario, I render only the most recent pinned post and then proceed as with #2. I also 
+wanted to figure out a way to render the excess pinned posts as usual, but I was not able to do so in time; since it was not a requirement, I decided 
+to table the matter for the time being in pursuit of avoiding making the perfect the enemy of the good.
+{%-endcomment-%}
+
+
 <div class="page-width">
   {% if section.settings.title != blank %}
     <header class="section-header text-center">
       <h2>{{ section.settings.title | escape }}</h2>
     </header>
   {% endif %}
+  
+  {%-comment-%} This is to prevent pinning from messing up the post limit. {%-endcomment-%}
+  {%- assign postLimit = section.settings.post_limit -%}
+  
+  {%-comment-%} This is to more than one article being pinned. {%-endcomment-%}
+  {%- assign pinnedRendered = false -%}
 
   {%- assign blog = blogs[section.settings.blog] -%}
 
   {% unless blog.articles_count == 0 %}
     <ul class="grid grid--uniform grid--blog">
-      {% for article in blog.articles limit: section.settings.post_limit %}
+      {% for article in blog.articles limit: postLimit %}
+      {% if article.tags contains "pinned" and pinnedRendered == false %}
+      	{%- decrement postLimit -%}
```

Comment:
Didn't think about decrementing the postLimit within the first for loop that checks for a pinned article. Logic is good.